cmd/kube-scheduler/app/options/options.go

```diff
@@ -121,8 +121,9 @@ func splitHostIntPort(s string) (string, int, error) {
 	if err != nil {
 		return "", 0, err
 	}
-	portInt, err := strconv.Atoi(port)
+	portInt, err := utilnet.ParsePort(port, false)
 	if err != nil {
+		klog.Errorf("Failed to parse endpoint port %s, error: %v", port, err)
```

Comment:
error logging and returning the error in parallel is suspicious, often an anti-pattern

Reply:
I will take out the logs.